HM2task1.c

```diff
@@ -0,0 +1,52 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+#define SIZE 47
+
+int recursivelyFibonacci(int num)
+{
+    if (num <= 1)
+    {
+        return 1;
+    }
+    return recursivelyFibonacci(num - 2) + recursivelyFibonacci(num - 1);
+}
+
+int iteractiveFibonacci(int num)
+{
+    int currentNum = 1;
+    int previousNum = 1;
+    for (int i = 2; i <= num; i++)
+    {
+        int temp = currentNum;
+        currentNum = previousNum + currentNum;
+        previousNum = temp;
+    }
+    return currentNum;
+}
+
+int main()
+{
+    int number = 0;
+    int scan_res = 0;
+    bool isIncorrectInput = true;
+
+    do
+    {
+        isIncorrectInput = true;
+
+        printf("Enter which Fibonacci number you want to output: ");
+        scan_res = scanf_s("%d", &number);
+
+        if (!scan_res || !isIncorrectInput || number < 0 || number >= SIZE)

```

Comment:
```suggestion
        if (!scan_res || number < 0 || number >= SIZE)
```
Эта переменная гарантировано `true`, так зачем её учитывать здесь?